### ISSUE TITLE
Support Pet feeder, Pet fountain

### ISSUE THREAD

**Issue by Semir333** [0x5e/homebridge-tuya-platform#92]
Hi there!

I seen, the Pet Feeder and Pet Fountain not supported by the plugin. I'm sucsessfully attached via the API, but can't do anything.
I've can help, with testing or grab informations if want because i have both devices (Petoneer Nutri Mini. Petoneer Fresco Ultra)

Also founded some other informations on other forums (this is by user: [hollowandi])

You can connect the petoneer with the localtuya integration!

Here is some assistance for the ultra version at least:

101 value:true power
102: value 10 indicator led dimming
103 value:133 change reminder water
104 value:133 change reminder filter
105 value: 7515 change reminder pump
106 value: 187 water TDS value or whatever it is
107 value:true UV on/off
108 value 2 water level
109 value 0 time for UV

I want to ask and help to integrate:

Pet Feeder | 宠物喂食器 | cwwsq 
Pet Fountain | 宠物饮水机 | cwysj

Thank you!

**Comment by 0x5e** [comment]
Thanks @Semir333 , I'm planing to add generic device support in 1.7.0, bring all control set displayed. The detail and design has not started. As the Pet Feeder / Fountain category can not be found in the [homebridge accessories](https://developers.homebridge.io/#/service), and it is not a normal or widely used device, I would consider treate them as generic device.

**Comment by 0x5e** [comment]
By the way, the number 101, 102 these are data point ID(dpId) which is used in lan protocol. In HTTP API we are using data point code(dpCode) such as `switch_led`, `brightness_value`. when you login into tuya platform and link your app account, they can be found in debugging page.

**Comment by Semir333** [comment]
Thank you very much! I waiting for it!

**Comment by davidliamfranks** [comment]
I also have a tuya pet feeder which would be great if we can get it working with some features? It appears in HomeKit but says "not supported"

**Comment by Juice805** [comment]
I'd also like to add category `sp` to this list which are the camera-enabled feeders. Some features such a light toggles work, but not the camera, nor a general "feed" function. 

Any improvements here would be great to see also brought to `sp` category

**Comment by bandlaw12** [comment]
Adding an upvote for this being supported in the future if possible. I have successfully added it to Homekit and would love the option to add a dummy trigger in homekit that would trigger an extra feeding and/or allow us to see the camera in Homekit. If there's anything I can do to help documentation wise please let me know!

**Comment by 0x5e** [comment]
@Juice805 I don't know if it works, before the "generic device support"

Sorry for the delay, here's the temp workaround:
For the camera-enabled feeders, try override the category to `sp`, I hope the camera can work (it depends on the cloud api). @Juice805 
For other features, please setup tuya tap-to-run on the app, restart homebridge, then it will appear as an switch. @bandlaw12 @Semir333

**Comment by bandlaw12** [comment]
Just tested and confirmed it works. Just for sake of anybody who needs to know, the tap-to-run is setup under "scene" from the app's main page, then you tap tap to run, then the plus button in top right. Select "Launch tap to run", and program "If Launch Tap-to-Run", then "Smart Feeder", "Manual Feeding: 1" (or whatever you want). Rebooted the child bridge, then restarted the homebridge server, and it showed up without further action in homekit in my default room as "smart feeder manual feeding:1"

Thank you @0x5e !

**Comment by Semir333** [comment]
Yes, it works, like before. But as i see, next tot the switches, the Homekit showes two accessory (both the feeder and water fountain) not supported. Is there able to remove it somehow? Btw, another question, is there any option to support the warnings, or mode displays? Eg. out of food warning, out of water warning (like a sensor), or UV mode active etc. etc. Thanks for the answer, and work @0x5e !
